engine/gamesys/src/gamesys/components/comp_camera.cpp

```diff
@@ -36,84 +36,78 @@ namespace dmGameSystem
     using namespace dmVMath;
 
     const uint32_t MAX_COUNT = 64;
-    const uint8_t MAX_STACK_COUNT = 8;
 
     struct CameraWorld;
 
     struct CameraComponent
     {
         dmGameObject::HInstance m_Instance;
-        CameraWorld* m_World;
-        float m_AspectRatio;
-        float m_Fov;
-        float m_NearZ;
-        float m_FarZ;
-        float m_OrthographicZoom;
-        dmVMath::Matrix4 m_View;
-        dmVMath::Matrix4 m_Projection;
-        uint32_t m_AutoAspectRatio : 1;
-        uint32_t m_AddedToUpdate : 1;
-        uint32_t m_OrthographicProjection : 1;
-        uint16_t m_ComponentIndex;
+        dmRender::HRenderCamera m_RenderCamera;
+        CameraWorld*            m_World;
+        dmVMath::Matrix4        m_View;
+        dmVMath::Matrix4        m_Projection;
+        uint16_t                m_ComponentIndex;
+        uint8_t                 m_AddedToUpdate : 1;
     };
 
     struct CameraWorld
     {
-        dmArray<CameraComponent> m_Cameras;
-        dmArray<CameraComponent*> m_FocusStack;
+        dmArray<CameraComponent>  m_Cameras;
+        dmArray<CameraComponent*> m_CameraStack;
     };
 
-    static const dmhash_t CAMERA_PROP_FOV = dmHashString64("fov");
-    static const dmhash_t CAMERA_PROP_NEAR_Z = dmHashString64("near_z");
-    static const dmhash_t CAMERA_PROP_FAR_Z = dmHashString64("far_z");
+    static const dmhash_t CAMERA_PROP_FOV               = dmHashString64("fov");
+    static const dmhash_t CAMERA_PROP_NEAR_Z            = dmHashString64("near_z");
+    static const dmhash_t CAMERA_PROP_FAR_Z             = dmHashString64("far_z");
     static const dmhash_t CAMERA_PROP_ORTHOGRAPHIC_ZOOM = dmHashString64("orthographic_zoom");
-    static const dmhash_t CAMERA_PROP_PROJECTION = dmHashString64("projection");
-    static const dmhash_t CAMERA_PROP_VIEW = dmHashString64("view");
-    static const dmhash_t CAMERA_PROP_ASPECT_RATIO = dmHashString64("aspect_ratio");
+    static const dmhash_t CAMERA_PROP_PROJECTION        = dmHashString64("projection");
+    static const dmhash_t CAMERA_PROP_VIEW              = dmHashString64("view");
+    static const dmhash_t CAMERA_PROP_ASPECT_RATIO      = dmHashString64("aspect_ratio");
 
 
-    void CompCameraUpdateViewProjection(CameraComponent* camera, dmRender::RenderContext* render_context)
+    static void CompCameraUpdateViewProjection(CameraComponent* camera, dmRender::RenderContext* render_context)
```

Comment:
Perhaps this should be a dmRender function? Since the renderer owns the render camera and its data now, perhaps we should expose this?

dmRender::UpdateRenderCamera(context, camera, world_position)